front/src/Phaser/Game/GameScene.ts

```diff
@@ -1081,8 +1082,76 @@ ${escapedMessage}
                 for (const eventTile of eventTiles) {
                     this.gameMap.putTile(eventTile.tile, eventTile.x, eventTile.y, eventTile.layer);
                 }
+                this.markDirty();
             })
         );
+        iframeListener.registerAnswerer("loadTileset", (eventTileset) => {
+            return this.connectionAnswerPromise.then(() => {
+                const jsonTilesetDir = eventTileset.url.substr(0, eventTileset.url.lastIndexOf("/"));
+                //Initialise the firstgid to 1 because if there is no tileset in the tilemap, the firstgid will be 1
+                let newFirstgid = 1;
+                const lastTileset = this.mapFile.tilesets[this.mapFile.tilesets.length - 1];
+                if (lastTileset) {
+                    //If there is at least one tileset in the tilemap then calculate the firstgid of the new tileset
+                    newFirstgid = lastTileset.firstgid + lastTileset.tilecount;
+                }
+                return new Promise((resolve, reject) => {
+                    this.load.on("filecomplete-json-" + eventTileset.url, () => {
+                        let jsonTileset = this.cache.json.get(eventTileset.url);
+                        this.load.image(
+                            jsonTilesetDir + "/" + jsonTileset.image,
+                            jsonTilesetDir + "/" + jsonTileset.image
+                        );
+                        this.load.on("filecomplete-image-" + jsonTilesetDir + "/" + jsonTileset.image, () => {
```

Comment:
repeat 3 in the code, you can create const of `jsonTilesetDir + "/" + jsonTileset.image`